habitat/tasks/rearrangement/rearrangement_task.py

```diff
@@ -0,0 +1,392 @@
+        distance_to_target = {}
+        habitat_sim = self._sim._sim
+        for i, sim_obj_id in enumerate(
+            habitat_sim.get_existing_object_ids()[1:]
+        ):
+            obj_id = habitat_sim.sim_object_to_objid_mapping[sim_obj_id]
+            previous_position = np.array(
+                habitat_sim.get_translation(sim_obj_id)
+            ).tolist()
+            agent_state = self._sim.get_agent_state()
+            agent_position = agent_state.position
+
+            distance_to_target[obj_id] = self._euclidean_distance(
+                previous_position, agent_position
+            )
+
+        self._metric = distance_to_target
+
+
+@registry.register_sensor
+class GrippedObjectSensor(Sensor):
+    def __init__(
+        self, *args: Any, sim: RearrangementSim, config: Config, **kwargs: Any
+    ):
+        self._sim = sim
+        super().__init__(config=config)
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        habitat_sim = self._sim._sim
+        return spaces.Discrete(habitat_sim.get_existing_object_ids())
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "gripped_object_id"
+
+    def _get_sensor_type(self, *args: Any, **kwargs: Any):
+        return SensorTypes.MEASUREMENT
+
+    def get_observation(
+        self,
+        observations: Dict[str, Observations],
+        episode: Episode,
+        *args: Any,
+        **kwargs: Any
+    ):
+        habitat_sim = self._sim._sim
+        obj_id = habitat_sim.sim_object_to_objid_mapping.get(
+            self._sim.gripped_object_id, -1
+        )
+        return obj_id
+
+
+@registry.register_sensor
+class AllObjectPositions(PointGoalSensor):
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "all_object_positions"
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        sensor_shape = (
+            5,
+            self._dimensionality,
+        )
+
+        return spaces.Box(
+            low=np.finfo(np.float32).min,
+            high=np.finfo(np.float32).max,
+            shape=sensor_shape,
+            dtype=np.float32,
+        )
+
+    def get_observation(
+        self, *args: Any, observations, episode, **kwargs: Any
+    ):
+        agent_state = self._sim.get_agent_state()
+        agent_position = agent_state.position
+        rotation_world_agent = agent_state.rotation
+        sensor_data = np.zeros((5, 2))
+        habitat_sim = self._sim._sim
+
+        for i, sim_obj_id in enumerate(
+            habitat_sim.get_existing_object_ids()[1:]
+        ):
+            obj_id = habitat_sim.sim_object_to_objid_mapping[sim_obj_id]
+            object_position = habitat_sim.get_translation(sim_obj_id)
+            sensor_data[obj_id] = self._compute_pointgoal(
+                agent_position, rotation_world_agent, object_position
+            )
+
+        return sensor_data
+
+
+@registry.register_sensor
+class AllObjectGoals(PointGoalSensor):
```

Comment:
Why do we inherit from PointGoalSensor, does not make sense to use a higher level abstract sensor? Like a Task_Sensor